lib/api/bulk.js

```diff
@@ -873,18 +873,26 @@ Bulk.prototype.query = function(soql) {
   var self = this;
   var recordStream = new RecordStream.Parsable();
   var dataStream = recordStream.stream('csv');
-  this.load(type, "query", soql).then(function(results) {
-    var streams = results.map(function(result) {
-      return self
+
+  this.load(type, "query", options, soql).then(function(results) {
+    var lazyLoadedResultStreams = results.map(function(result) {
+      const batch = self
         .job(result.jobId)
-        .batch(result.batchId)
-        .result(result.id)
-        .stream();
+        .batch(result.batchId);
+
+      return function() {
```

Comment:
@david0673 can you explain more how this 'avoid timeouts from idle open connections'? Is this because `multistream` operates on one at a time so by lazy loading we don't kick off the streaming until we are actually processing them?

Thanks for digging into this!

Reply:
Exactly